crates/config/src/sections/upstream_oauth2.rs

```diff
@@ -401,6 +421,14 @@ pub struct Provider {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub token_endpoint_auth_signing_alg: Option<JsonWebSignatureAlg>,
 
+    /// Whether to fetch the user profile from the userinfo endpoint,
+    /// or to rely on the data returned in the id_token from the token_endpoint.
+    ///
+    /// Defaults to `auto`, which uses the userinfo endpoint if `openid` is not
+    /// included in `scopes`, and the ID token otherwise.
+    #[serde(default, skip_serializing_if = "UserProfileMethod::is_default")]
+    pub user_profile_method: UserProfileMethod,
```

Comment:
I'm not convinced we should replicate the Synapse behaviour here, to avoid implicit behaviour as much as possible.

What I would like to have instead, is just a boolean whether we do fetch the userinfo or not; this way, we always get/verify the ID token if we get one, we fetch the userinfo if it is configured to, and we make both available in the template, with the nice merge you did for the `user` property